presto-main/src/main/java/com/facebook/presto/type/UnknownType.java

```diff
@@ -32,6 +32,44 @@ private UnknownType()
     }
 
     @Override
+    public boolean isComparable()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean isOrderable()
+    {
+        return true;
+    }
+
+    @Override
+    public int hash(Block block, int position)
+    {
+        // Check that the position is valid
+        block.isNull(position);
+        return 0;
+    }
+
+    @Override
+    public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        // Check that the position is valid
+        leftBlock.isNull(leftPosition);
+        rightBlock.isNull(rightPosition);
+        return true;
```

Comment:
This is only valid if the values of the unknown type are NULL, since we can't otherwise compare them. So, make sure to enforce this assumption to avoid masking errors in the type system.